lib/Remotery.c

```diff
@@ -621,31 +622,42 @@ static void mtxDelete(rmtMutex* mutex)
 // be used to update the old value and an initial load only made once before the loop starts.
 
 // TODO(don): Vary these types across versions of C and C++
+#ifdef RMT_PLATFORM_MACOS
+typedef _Atomic(rmtS32) rmtAtomicS32;
+typedef _Atomic(rmtU32) rmtAtomicU32;
+typedef _Atomic(rmtU64) rmtAtomicU64;
+typedef _Atomic(void*) rmtAtomicPtr;
+#else
 typedef volatile rmtS32 rmtAtomicS32;
 typedef volatile rmtU32 rmtAtomicU32;
 typedef volatile rmtU64 rmtAtomicU64;
+#endif
 
-static rmtBool AtomicCompareAndSwapU32(rmtU32 volatile* val, long old_val, long new_val)
+static rmtBool AtomicCompareAndSwapU32(rmtAtomicU32 volatile* val, rmtU32 old_val, rmtU32 new_val)
```

Comment:
Is there any reason you're preserving the `volatile` in this change? See `AtomicCompareAndSwapU64`.

Reply:
no.
I thought it might be useful on other platforms to keep it this way, but apart from that, `volatile` might be removed altogether...

Reply:
No, if it compiles fine on Mac it'll be good. I'll change all that on my end.